src/lazy/any_encoding.rs

```diff
@@ -67,7 +65,7 @@ pub struct AnyEncoding;
 // underlying type.
 impl Decoder for AnyEncoding {
     type Reader<'data> = LazyRawAnyReader<'data>;
-    type ReaderSavedState = RawReaderType;
+    type ReaderSavedState = IonEncoding;
```

Comment:
🗺️ This type (an enum of the four Ion encodings) is now returned by the new (feature gated) `SystemReader<AnyEncoding, _>::detected_encoding()` method, allowing users to determine what encoding is currently being read.